internal/binary/binary.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package binary provides support for operating on binary data.
+package binary
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+)
+
+// ExpectBytes reads len(want) bytes from r and returns an error on mismatch.
+func ExpectBytes(r io.Reader, want []byte) error {
+	got := make([]byte, len(want))
+	if _, err := io.ReadFull(r, got); err != nil {
+		return fmt.Errorf("reading magic: %v", err)
```

Comment:
This doesn't know about magic anymore, so just

fmt.Errorf("reading bytes: %v", err)

[and fmt.Errorf("comparing bytes: expected %v not %v", got, want)

Reply:
done